x-pack/plugins/observability_solution/observability_ai_assistant_management/public/routes/components/settings_tab/product_doc_entry.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState, useCallback, useMemo } from 'react';
+import {
+  EuiButton,
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHealth,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useKibana } from '../../../hooks/use_kibana';
+import { useGetProductDocStatus } from '../../../hooks/use_get_product_doc_status';
+import { useInstallProductDoc } from '../../../hooks/use_install_product_doc';
+import { useUninstallProductDoc } from '../../../hooks/use_uninstall_product_doc';
+
+export function ProductDocEntry() {
+  const {
+    overlays,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  const [isInstalled, setInstalled] = useState<boolean>(true);
+  const [isInstalling, setInstalling] = useState<boolean>(false);
+
+  const { mutateAsync: installProductDoc } = useInstallProductDoc();
+  const { mutateAsync: uninstallProductDoc } = useUninstallProductDoc();
+  const { status, isLoading: isStatusLoading } = useGetProductDocStatus();
+
+  useEffect(() => {
+    if (status) {
+      setInstalled(status.overall === 'installed');
+    }
+  }, [status]);
+
+  const onClickInstall = useCallback(() => {
```

Comment:
If the install fails I get the error toast, but the UI remains with "(spinner) Installing...".  Perhaps the state should be reset so they can try again if desired.